src/main.ts

```diff
@@ -0,0 +1,238 @@
+import * as fs from 'fs';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import * as exec from '@actions/exec';
+import * as cache from '@actions/cache';
+import got from 'got';
+
+export enum Inputs {
+  CurrentStatus = 'current-status',
+  SlackWebhook = 'slack-webhook',
+  GithubToken = 'github-token',
+}
+
+const { context } = github;
+const repository = context.repo.repo;
+const cachePrimaryKey = `last-run-status-${context.runId}-${Math.random()
+  .toString(36)
+  .substr(2, 12)}`;
+const cacheRestoreKeys = [`last-run-status-${context.runId}-`];
+const cachePaths = ['last-run-status'];
+
+/**
+ * Gets the last workflow run status
+ *
+ * @returns last run status
+ */
+async function getLastRunStatus() {
+  let lastStatus = '';
+
+  const cacheKey = await cache.restoreCache(
+    cachePaths,
+    cachePrimaryKey,
+    cacheRestoreKeys,
+  );
+
+  if (!cacheKey || (cacheKey && !fs.existsSync(cachePaths[0]))) {
+    core.info('Cache not found, retrieve status from previous run.');
+
+    let headRef;
+
+    if (context.payload.pull_request !== undefined) {
+      headRef = JSON.parse(JSON.stringify(context.payload.pull_request)).head
+        .ref;
+    } else {
+      headRef = context.ref.split('/').pop();
+    }
+
+    core.info(`Branch name: ${headRef}`);
+
+    const githubToken = core.getInput(Inputs.GithubToken);
+    core.exportVariable('GITHUB_TOKEN', `${githubToken}`);
+
+    const options: any = {};
+    options.listeners = {
+      stdout: (data: Buffer) => {
+        lastStatus += data.toString();
+      },
+    };
+
+    await exec.exec(
+      '/bin/bash',
+      [
+        '-c',
+        `gh run list -w "${context.workflow}" | grep "${context.workflow}	${headRef}" | grep -v "completed	cancelled" | grep -v "in_progress" | head -n 1 | awk -F" " '{print $1"/"$2}'`,
+      ],
+      options,
+    );
+
+    core.info(`GH Found status: ${lastStatus}`);
+  } else {
+    core.info('Cache found, retrieve status from same run.');
+
+    lastStatus = fs.readFileSync(cachePaths[0], 'utf8');
+
+    core.info(`Cache Found status: ${lastStatus}`);
+  }
+
+  return lastStatus.trim();
+}
+
+/**
+ * Prepare slack notification
+ *
+ * @param webhookURL
+ * @param messageBody
+ * @param message
+ * @param status
+ * @returns the Slack message body
+ */
+async function prepareSlackNotification(
+  message: string,
+  status: string,
+): Promise<Record<string, any>> {
+  const { runId } = context;
+  const event = context.eventName;
+  const { owner } = context.repo;
+  const { workflow } = context;
+  const { actor } = context;
+  const { serverUrl } = context;
+  const color = status === 'success' ? 'good' : 'danger';
+
+  let headRef;
+  if (context.payload.pull_request !== undefined) {
+    headRef = JSON.parse(JSON.stringify(context.payload.pull_request)).head.ref;
```

Comment:
Same comment as above. Maybe worth being it's own function since it's the same logic.